ivadomed/utils.py

```diff
@@ -236,14 +239,34 @@ def structurewise_uncertainty(fname_lst, fname_hard, fname_unc_vox, fname_out):
         nib_im = nib.load(fname)
         data_im = nib_im.get_fdata()
         data_lst.append(data_im)
-        data_im_l, _ = label(data_im, structure=bin_struct)
+        data_im_l = []
+        for i in range(data_im.shape[-1]):
+            data_im_tmp, _ = label(data_im[..., i], structure=bin_struct)
```

Comment:
Not sure: do we need a `np.array(bin_struct)` as in L225? which can also be put directly at L223 I think, if needed.